client/trino-client/src/main/java/io/trino/client/ProtocolHeaders.java

```diff
@@ -190,8 +191,11 @@ public static ProtocolHeaders detectProtocol(Optional<String> alternateHeaderNam
         requireNonNull(headerNames, "headerNames is null");
 
         if (alternateHeaderName.isPresent() && !alternateHeaderName.get().equalsIgnoreCase("Trino")) {
-            if (headerNames.contains("X-" + alternateHeaderName.get() + "-User")) {
-                if (headerNames.contains("X-Trino-User")) {
+            String headerPrefix = "x-" + alternateHeaderName.get().toLowerCase(ENGLISH);
+            if (headerNames.stream()
```

Comment:
Nit: these would look better without wrapping

Reply:
You mean something like:

```
            if (headerNames.stream().anyMatch(header -> header.toLowerCase(ENGLISH).startsWith(headerPrefix))) {
                if (headerNames.stream().anyMatch(header -> header.toLowerCase(ENGLISH).startsWith("x-trino-"))) {
```

Reply:
Yep, exactly